feat: Add product delete.

Thiagosavin opened the pull request:
Issue: #2 

### What was done
Added DELETE /v1/products route
### Why
<!-- 
Justifique o motivo da mudança. 
Qual problema está sendo resolvido? Qual valor isso traz para o projeto?
Exemplo: "To allow admins to add products through the API", "To improve performance on the product listing page", etc.
-->
To allow admin exclude a product from list
### How to test
<!-- 
Liste os passos necessários para testar essa PR.
Inclua testes locais, testes automatizados e qualquer verificação adicional.
Use uma checklist clara e objetiva.
-->
1. By swagger doc in admin section
2. Unit and Integration tests with ```npm run test```command

Thiagosavin committed:
feat: Implement soft delete functionality for products.

dviniciusbonin commented on src/core/product/application/ports/product-repository.ts:
```diff
@@ -4,4 +4,5 @@ export abstract class ProductRepository {
   abstract create(product: Product): Promise<void>;
   abstract findById(id: string): Promise<Product | null>;
   abstract update(product: Product): Promise<void>;
+  abstract softDelete(id: string): Promise<void>;
```
Acho que no contrato do repositório faria mais sentido usar um nome genérico tipo delete em vez de softDelete. A camada de aplicação não precisa saber se é soft ou hard delete — isso é detalhe da implementação.

Thiagosavin committed:
refactor: Rename softDelete method to delete in product repository interfaces and implementations

Thiagosavin committed:
feat: Implement pagination for client listing and add related DTOs and factories

dviniciusbonin reviewed (approved):
Approved!

Thiagosavin committed:
feat: restructure order domain to use Item entity instead of OrderProduct

- Removed OrderProduct entity and replaced it with Item entity for better clarity and functionality.
- Added validation for item quantity in Item entity.
- Updated Order entity to manage items and calculate total based on Item instances.
- Introduced DuplicatedItemError for handling duplicate items in an order.
- Enhanced order creation and retrieval processes to accommodate new item structure.
- Updated related tests to reflect changes in order and item handling.
- Added WaitingTime value object to calculate and format waiting times for orders.
- Refactored API DTOs to align with new item structure and naming conventions.

Thiagosavin committed:
Merge branch 'main' of github.com:postech-group-fiap-188/fastfood-system into (@thiagosavin/4/product-remove
